python/pyarrow/table.pxi

```diff
@@ -77,6 +77,49 @@ cdef class ChunkedArray:
         self._check_nullptr()
         return self.chunked_array.null_count()
 
+    def __getitem__(self, key):
+        cdef int64_t item, i
+        self._check_nullptr()
+        if isinstance(key, slice):
+            return _normalize_slice(self, key)
+        else:
+            item = int(key)
+            if item >= self.chunked_array.length() or item < 0:
+                return IndexError("ChunkedArray selection out of bounds")
+            for i in range(self.num_chunks):
```

Comment:
I'm curious, can't this be implemented on the C++ side instead? I suppose C++ users may benefit from such an API as well.

Reply:
Yes they would benefit from this API but it is much more complicated to implement there as `arrow::ChunkedArray` has no type.